src/burn/engine/plan.cpp

```diff
@@ -2775,6 +2797,22 @@ static HRESULT PlanDependencyActions(
     hr = DependencyPlanPackageBegin(fBundlePerMachine, pPackage, pPlan);
     ExitOnFailure(hr, "Failed to begin plan dependency actions for package: %ls", pPackage->sczId);
 
+    // All packages that have cacheType set to always should be cached if the bundle is going to be present.
```

Comment:
I put this block of code in different places than your original pull request, adding it here will cause it to be done twice.  The `BURN_CACHE_TYPE_ALWAYS` condition is in `ProcessPackage` and `NeedsCache`.  `AddCacheSlipstreamMsps` is in `AddCachePackage`.